test/Constraints/generics.swift

```diff
@@ -631,3 +631,11 @@ let layout = BottleLayout(count:1)
 let ix = arr.firstIndex(of:layout) // expected-error {{argument type 'BottleLayout' does not conform to expected type 'Equatable'}}
 
 let _: () -> UInt8 = { .init("a" as Unicode.Scalar) } // expected-error {{initializer 'init(_:)' requires that 'Unicode.Scalar' conform to 'BinaryInteger'}}
+
+// rdar://problem/45659733
+func rdar_45659733() {
```

Comment:
Maybe this belongs more in test/Constraints/closures.swift, but I'm not too picky.

Reply:
You are right, I think closures might be better place for this!